localization/landmark_based_localizer/ar_tag_based_localizer/src/ar_tag_based_localizer.cpp

```diff
@@ -305,37 +301,18 @@ void ArTagBasedLocalizer::cam_info_callback(const CameraInfo::ConstSharedPtr & m
 
 void ArTagBasedLocalizer::ekf_pose_callback(const PoseWithCovarianceStamped::ConstSharedPtr & msg)
 {
-  // lock mutex for initial pose
-  std::lock_guard<std::mutex> self_pose_array_lock(self_pose_array_mtx_);
-  // if rosbag restart, clear buffer
-  if (!self_pose_msg_ptr_array_.empty()) {
-    const builtin_interfaces::msg::Time & t_front = self_pose_msg_ptr_array_.front()->header.stamp;
-    const builtin_interfaces::msg::Time & t_msg = msg->header.stamp;
-    if (t_front.sec > t_msg.sec || (t_front.sec == t_msg.sec && t_front.nanosec > t_msg.nanosec)) {
-      self_pose_msg_ptr_array_.clear();
-    }
+  if (ekf_pose_buffer_->detect_time_jump_to_past(msg->header.stamp)) {
```

Comment:
Please explain why this check exists. We should make sure that future viewers of this code will know why this process is necessary. In my understand, if the system is normal, the clock should not go back to the past.
e.g. `// This is to ensure that if rosbag is restarted, the process will continue.`